cw_infrastructure/cw_infrastructure/check_config.py

```diff
@@ -55,39 +54,21 @@ def get_file_name(path):
     return file_name
 
 
-def import_option_modules():
-    """Retrieve and import option modules from the options/ directory"""
-
-    # Search the specified directory for python modules and obtain (filename, path)
-    # pairs for each such module.
-    options_path = os.path.join(OPTION_MODULE_PATH, '*.py')
-    option_pairs = [(get_file_name(path), path) for path in glob(options_path)]
-
-    # Import the modules, and return a list containing them
-    option_modules = [imp.load_source(name, path) for (name, path) in option_pairs]
-    return option_modules
-
-
-if __name__ == '__main__':
-    # Retrieve and import option modules
-    option_modules = import_option_modules()
+def check_config():
+    status = OK
 
-    # Build up the list of options to be validated
-    options = []
-    for module in option_modules:
-        options += module.get_options()
+    # Build up a list of checks to be performed. Each check should take no
+    # arguments and return a status code.
+    checks = [functools.partial(check_config_options,
+                                clearwater_options.get_options()),
+              clearwater_options.check_advanced_config]
 
-    status = OK
+    # Determine the resultant status code - since ERROR > WARNING > OK, take
+    # the maximum.
+    status = max(check() for check in checks)
```

Comment:
You can just write `return max(check() for check in checks)`, no need to store it in a variable first.

(In Python people tend to be OK with this sort of thing because Python encourages you to be more flexible with your return statements - when you can have multiple return statements in a function, there's less need to be as rigid with how they are instantiated).